src/overlays/actors/ovl_Bg_Spot16_Doughnut/z_bg_spot16_doughnut.c

```diff
@@ -21,15 +30,136 @@ const ActorInit Bg_Spot16_Doughnut_InitVars = {
+    ICHAIN_F32(unk_F8, 5000, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_FC, 5000, ICHAIN_STOP),
+};
+
+static s16 sScales[] = {
+    0, 0, 70, 210, 300,
+};
+
+extern Gfx D_06000660[];
+extern Gfx D_06000FC0[];
+
+void BgSpot16Doughnut_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot16Doughnut* this = THIS;
+    s32 params;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Actor_SetScale(&this->actor, 0.10000000149011612f);
+    this->fireFlag = 0;
+    this->envColorAlpha = 0xFF;
+    params = this->actor.params;
+    if (params == 1 || params == 2 || params == 3 || params == 4) {
+        Actor_SetScale(&this->actor, sScales[this->actor.params] * 9.999999747378752e-05f);
+        this->actor.draw = BgSpot16Doughnut_Draw2;
+        this->actor.update = BgSpot16Doughnut_Update2;
+    } else {
+        // Scales this actor for scenes where it is featured in the background,
+        // Death Mountain itself falls into the default case.
+        switch (globalCtx->sceneNum) {
+            case SCENE_SPOT01:
+                Actor_SetScale(&this->actor, 0.03999999910593033f);
+                break;
+            case SCENE_SHRINE:
+            case SCENE_SHRINE_N:
+            case SCENE_SHRINE_R:
+                Actor_SetScale(&this->actor, 0.017999999225139618f);
+                break;
+            default:
+                Actor_SetScale(&this->actor, 0.10000000149011612f);
+                break;
+        }
+        osSyncPrintf(VT_FGCOL(CYAN) "%f" VT_RST "\n", this->actor.scale.x);
+        if (LINK_IS_CHILD || gSaveContext.eventChkInf[2] & 0x8000) {
+            this->fireFlag &= ~1;
+        } else {
+            this->fireFlag |= 1;
+        }
+        osSyncPrintf("(ｓｐｏｔ１６ ドーナツ雲)(arg_data 0x%04x)\n", this->actor.params);
+    }
+}
+
+void BgSpot16Doughnut_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void BgSpot16Doughnut_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot16Doughnut* this = THIS;
+
+    if (!(this->fireFlag & 1)) {
+        this->actor.shape.rot.y -= 0x20;
+        if (this->envColorAlpha < 0xFF) {
+            this->envColorAlpha += 5;
+        } else {
+            this->envColorAlpha = 0xFF;
+        }
+    } else if (globalCtx->csCtx.state != 0 && globalCtx->csCtx.npcActions[2] != NULL &&
+               (u16)globalCtx->csCtx.npcActions[2]->action == 2) {
+        if (this->envColorAlpha >= 6) {
+            this->envColorAlpha -= 5;
+        } else {
+            this->envColorAlpha = 0;
+            this->fireFlag &= ~1;
+        }
+    }
+}
+
+// Update function for outwardly expanding and dissipating
+void BgSpot16Doughnut_Update2(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot16Doughnut* this = THIS;
+
+    if (this->envColorAlpha >= 6) {
+        this->envColorAlpha -= 5;
+    } else {
+        Actor_Kill(&this->actor);
+    }
+    this->actor.shape.rot.y -= 0x20;
+    Actor_SetScale(&this->actor, this->actor.scale.x + 0.001999999862164259f);
```

Comment:
```suggestion
    Actor_SetScale(&this->actor, this->actor.scale.x + 0.002f);
```

Reply:
This one didn't match but I was able to shorten it some